72_simplify_tls.md

```diff
@@ -0,0 +1,233 @@
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [keylime/enhancements] referencing this enhancement and targeting a release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Keylime i.e., [keylime/keylime], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+- [ ] Enhancement issue in release milestone, which links to pull request in
+  [keylime/enhancements]
+- [ ] Core members have approved the issue with the label `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place
+- [ ] User-facing documentation has been created in [keylime/keylime-docs]
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+## Summary
+
+The current TLS setup uses one CA for the verifier, one for the registrar and
+one for the agent mTLS authentication. This will be replaced by a client and
+server (key and certificate) option in each component (agent, verifier,
+registrar, tenant). The client pair is used to authenticate the component using
+mTLS and the server pair is used for TLS server (like the registrar, agent and
+verifier have). Also two new options are introduced, one for trustworthy
+incoming CAs for server side mTLS authentication and another one for trustworthy
+CAs from the client side.
+
+## Motivation
+
+The current way TLS is setup in Keylime is confusing to newcomers. Also the
+different configuration options are not consistent in their naming scheme.
+
+
+### Goals
+
+* Simplify the TLS setup of Keylime
+* Cleanup keylime_ca to make it work with the new setup
+
+### Non-Goals
+
+* Implement other mechanisms of authentication other than mTLS
+* Move registration to HTTPS (but can be easily done in a followup change)
+
+## Proposal
+To the TLS related options in Keylime we will make the following changes.
+
+### New options
+The old options are replaced with the following new ones:
+
+* `tls_dir`: Path to where the files are located
+* `client_key`: Key for mTLS client connections
+* `client_cert`: Certificate for mTLS client connections
+* `sever_key`: Key for a TLS server
+* `server_cert`: Certificate for a TLS server
+* `trusted_ca_in`: List of trusted CAs for incoming connections
+* `trusted_ca_out`: List of trusted CAs for outgoing connections
```

Comment:
why we should separate in and out?

Reply:
You want to trust the CA of the tenant and the verifier for incoming connections as an agent, but not the registrar CA.
In the future you still want to validate the CA of registrar if you make connections to it.

Reply:
I changed it to `trusted_{client,server}_ca` for more clarity, as suggested.